lib/mqtt.c

```diff
@@ -441,7 +441,7 @@ static CURLcode mqtt_read_publish(struct Curl_easy *data, bool *done)
   size_t remlen;
   struct mqtt_conn *mqtt = &conn->proto.mqtt;
   struct MQTT *mq = data->req.p.mqtt;
-  unsigned char packet;
+  int packet;
```

Comment:
Why this change? It only stores a byte/char in there?

Reply:
Again, this was just stuff my linter picked up and I changed the type pretty blindly to make the linter play nice.

Seeing as you're actually open to accepted this PR, I'm putting some more time and effort in now to get it up to scratch, and ensure all the tests pass, and that the total number of warnings is reduced.